mlir/lib/Target/LLVMIR/Dialect/GENX/GENXToLLVMIRTranslation.cpp

```diff
@@ -40,6 +41,19 @@ static llvm::Value *createDeviceFunctionCall(llvm::IRBuilderBase &builder,
   return builder.CreateCall(fn, args);
 }
 
+// Create a call to SPIR atomic cmpxchg function.
+static llvm::Value *createAtomicCmpXchg(llvm::IRBuilderBase &builder,
+                                        llvm::Value *ptr, llvm::Value *cmp,
+                                        llvm::Value *val) {
+  assert(isa<llvm::PointerType>(ptr->getType()) && "Expecting a pointer type");
+  assert(cmp->getType() == val->getType() && "Mismatching types");
```

Comment:
should we assert ptr element type is the same as val type?

Reply:
No because ptr are opaque.

Reply:
ok